### ISSUE TITLE
Missing methods when generating

### ISSUE THREAD

**Issue by nisbus** [tomas-abrahamsson/gpb#192]
I have a file with a lot of enums defined but 4 of them get converted badly (the from number method is undefined).  
  
I have this enum:  
```proto
enum LogLevel {
	LOGLEVEL_EMERGENCY  = 0;
	LOGLEVEL_ALERT      = 1;
	LOGLEVEL_CRITICAL   = 2;
	LOGLEVEL_ERROR      = 3;
	LOGLEVEL_WARNING    = 4;
	LOGLEVEL_NOTICE     = 5;
	LOGLEVEL_INFO       = 6;
	LOGLEVEL_DEBUG      = 7;
	LOGLEVEL_DEBUG2     = 8;
}
```
  
```erlang
-compile({nowarn_unused_function,fj_enum_LogLevel/1}).
fj_enum_LogLevel(S) when is_binary(S) ->
    case fj_casecanon_enum(S, <<>>) of
      <<"LOGLEVEL_EMERGENCY">> -> 'LOGLEVEL_EMERGENCY';
      <<"LOGLEVEL_ALERT">> -> 'LOGLEVEL_ALERT';
      <<"LOGLEVEL_CRITICAL">> -> 'LOGLEVEL_CRITICAL';
      <<"LOGLEVEL_ERROR">> -> 'LOGLEVEL_ERROR';
      <<"LOGLEVEL_WARNING">> -> 'LOGLEVEL_WARNING';
      <<"LOGLEVEL_NOTICE">> -> 'LOGLEVEL_NOTICE';
      <<"LOGLEVEL_INFO">> -> 'LOGLEVEL_INFO';
      <<"LOGLEVEL_DEBUG">> -> 'LOGLEVEL_DEBUG';
      <<"LOGLEVEL_DEBUG2">> -> 'LOGLEVEL_DEBUG2';
      <<"0">> -> 'LOGLEVEL_EMERGENCY';
      <<"1">> -> 'LOGLEVEL_ALERT';
      <<"2">> -> 'LOGLEVEL_CRITICAL';
      <<"3">> -> 'LOGLEVEL_ERROR';
      <<"4">> -> 'LOGLEVEL_WARNING';
      <<"5">> -> 'LOGLEVEL_NOTICE';
      <<"6">> -> 'LOGLEVEL_INFO';
      <<"7">> -> 'LOGLEVEL_DEBUG';
      <<"8">> -> 'LOGLEVEL_DEBUG2';
      _ -> 'LOGLEVEL_EMERGENCY'
    end;
fj_enum_LogLevel(N) when is_integer(N) -> d_enum_LogLevel(N);
fj_enum_LogLevel(S) when is_list(S) -> fj_enum_LogLevel(list_to_binary(S)).
```    
this all looks kosher except the `d_enum_LogLevel(N)` method is not defined anywhere.    
    
On the other hand I have another enum:  
```proto
enum Severity {
  SEVERITY_INFO = 0;
  SEVERITY_WARN = 1;
  SEVERITY_ERROR = 2;
}
```  
which works fine:
  
```erlang
-compile({nowarn_unused_function,fj_enum_Severity/1}).
fj_enum_Severity(S) when is_binary(S) ->
    case fj_casecanon_enum(S, <<>>) of
      <<"SEVERITY_INFO">> -> 'SEVERITY_INFO';
      <<"SEVERITY_WARN">> -> 'SEVERITY_WARN';
      <<"SEVERITY_ERROR">> -> 'SEVERITY_ERROR';
      <<"0">> -> 'SEVERITY_INFO';
      <<"1">> -> 'SEVERITY_WARN';
      <<"2">> -> 'SEVERITY_ERROR';
      _ -> 'SEVERITY_INFO'
    end;
fj_enum_Severity(N) when is_integer(N) -> d_enum_Severity(N);
fj_enum_Severity(S) when is_list(S) -> fj_enum_Severity(list_to_binary(S)).
```
  
```erlang
d_enum_Severity(0) -> 'SEVERITY_INFO';
d_enum_Severity(1) -> 'SEVERITY_WARN';
d_enum_Severity(2) -> 'SEVERITY_ERROR';
d_enum_Severity(V) -> V.
```  
The resulting file is over 30k lines so I can't share it or the proto files, sorry.  
Any ideas?  

Thanks,  
nisbus

**Comment by nisbus** [comment]
I'm using the `fix-extend-with-proto3-syntax` branch

**Comment by tomas-abrahamsson** [comment]
It looks like this could happen when there are enums that no message field has as type. Could this be the case for in your protos, ie that no message field is of type `LogLevel`?

If so, then I think I have a fix. For your convenience, I pushed it again to the `fix-extend-with-proto3-syntax` branch, since I figure you need both corrections. When I merge I will restructure the branches to relect that the changes are independent.

**Comment by nisbus** [comment]
This fixes it also.
The thing is that I was only generating a part of the protos since that's all I needed for this project.
The LogLevel is used but not in the files being referenced there.

Thanks a lot for your extraordinary quick response

You rock!
nisbus

**Comment by tomas-abrahamsson** [comment]
Haha, thanks! :) I'll merge borh fixes into a new release soon.

**Comment by tomas-abrahamsson** [comment]
The fix is merged and landed in 4.16.0
